### PR TITLE
[CBRD-22950] fix replication_channels unit test compile & test

### PR DESCRIPTION
**Pull request by radudoros**
http://jira.cubrid.org/browse/CBRD-22950

Fix slip of https://github.com/CUBRID/cubrid/pull/1609 (compile issue) & test behavior.

### CODE REVIEW

File: unit_tests/common/mock_stream.cpp [hunk 1]
```diff
@@ -3,9 +3,14 @@
 #include <assert.h>
 #include <error_code.h>
 
+namespace cubcomm
```
Comment (ribram) [inline unit_tests/common/mock_stream.cpp]: Isn't MTU already defined in `communication_channel.hpp`?

File: unit_tests/common/mock_stream.cpp [hunk 2]
```diff
@@ -23,7 +25,7 @@ int mock_stream::write (const size_t byte_count, cubstream::stream::write_func_t
 {
   int err;
 
-  err = write_action (last_position, write_buffer, byte_count);
+  err = write_action (last_position, write_buffer + last_position, byte_count);
```
Comment (razvanradulescu) [inline unit_tests/common/mock_stream.cpp]: I think the old code was correct.

Comment (radudoros) [reply unit_tests/common/mock_stream.cpp]: well, the old code would overwrite the same last written part of the buffer and implicitly would mess up the postcondition of the test, which checks that the sum is n(n-1)/2.

Comment (razvanradulescu) [reply unit_tests/common/mock_stream.cpp]: I accept that this is the logic of test.

**Pull request merged**